roles/ceph-common/handlers/main.yml

```diff
@@ -1,8 +1,12 @@
 ---
+# upgrading ceph from hammer to jewel will make the restarting failed,
+# this will cause ceph cluster down.
+# this happens when operater really upgrades ceph and forget to set ceph.upgrade2jewel=true
+# so we can't set default value of ceph.upgrade2jewel as false here
 - name: restart ceph mons
   service: name=ceph-mon-all state=restarted
-  when: ceph_socket.rc == 0 and not ceph.scaleout|default('False')|bool
+  when: ceph_socket.rc == 0 and not ceph.scaleout|default('False')|bool and not ceph.upgrade2jewel|bool
 
 - name: restart ceph osds
   service: name=ceph-osd-all state=restarted
-  when: ceph_socket.rc == 0 and not ceph.scaleout|default('False')|bool
+  when: ceph_socket.rc == 0 and not ceph.scaleout|default('False')|bool and not ceph.upgrade2jewel|bool
```

Comment:
Why do we need a new flag ceph.upgrade2jewel. Cannot we rely on upgrade flag? 
Upgrading to Jewel is not optional.